### PR TITLE
Add MFT connector

### PR DESCRIPTION
**Pull request by SkNuwanTissera**
# Description

Add MFT connector https://mftgateway.com/
Feature https://github.com/wso2-enterprise/choreo/issues/11526

One line release note: 
- Add MFT connector

## Type of change

- [X] New feature (non-breaking change which adds functionality)

**Test Configuration**:
* Ballerina Version: 2201.0.0
* Operating System: Ubuntu
* Java SDK: Java11

# Checklist:

### Changes to OpenAPI definition

- [x] Modified `info` section of the OpenAPI definition file - [Example](https://github.com/ballerina-platform/ballerina-extended-library/discussions/74)

    - `description` - what the connector is about 
    - `x-ballerina-init-description` OpenAPI extension - connector initialization details 
    - `x-ballerina-display` OpenAPI extension - connector name and path to the connector icon 

- [x] Added proper description each API Key, if API Key authentication is defined in OpenAPI definition file 

### Changes to connector module

- [x] Added license header at the top of each .bal file. 
- [x] Added Package.md as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/77)
- [x] Added Module.md for the module of the  connector as per the guide [here](https://github.com/ballerina-platform/ballerina-extended-library/discussions/78)
- [x] Added connector icon to the root of the connector. Icon needs to be a `png` of 200x200 px size with name `icon.png`
- [x] Added Ballerina.toml file with following information. For keywords refer to [guide](https://github.com/ballerina-platform/ballerina-extended-library/discussions/72)

### Verifying connector module

- [x] Compiled the connector successfully with the targeted ballerina version. 
- [x] Conducted smoke tests on the connector (Recommended when OpenAPI definition is obtain from an unofficial source)

### Security checks
 - [x] Followed secure coding standards in http://wso2.com/technical-reports/wso2-secure-engineering-guidelines? 
 - [x] Confirmed that this PR doesn't commit any keys, passwords, tokens, usernames, or other secrets?

### CODE REVIEW

File: openapi/mft/Ballerina.toml [hunk 1]
```diff
@@ -0,0 +1,13 @@
+[package]
+license = ["Apache-2.0"]
+keywords = ["IT Operations/Message Brokers", "Cost/Freemium"]
+org = "choreo"
```
Comment (LakshanSS) [inline openapi/mft/Ballerina.toml:4]: Shouldn't this be `ballerinax`?

Comment (SkNuwanTissera) [reply openapi/mft/Ballerina.toml]: Ah yes. Thanks 
Updated

File: openapi/mft/utils.bal [hunk 2]
```diff
@@ -0,0 +1,277 @@
...
+        entities.push(entity);
+    }
+    return entities;
+}
+
+# Encoding AS2identifier
+#
+# + as2Identifier - unencoded as2Identifier (e.g: <333900490041112@mftgateway.com> )
+# + return - Encoded string or error
+isolated function encodePathParam(string as2Identifier) returns string|error {
```
Comment (aneeshafedo) [inline openapi/mft/utils.bal:275]: Cant we reuse  `getEncodedUri` function since it does the same thing.

Comment (SkNuwanTissera) [reply openapi/mft/utils.bal]: Updated with changes

File: openapi/mft/Ballerina.toml [hunk 3]
```diff
@@ -0,0 +1,13 @@
+[package]
+license = ["Apache-2.0"]
+keywords = ["IT Operations/Message Brokers", "Cost/Freemium"]
+org = "ballerinax"
+name = "mft"
+icon = "icon.png"
+distribution = "2201.0.0"
+repository = "https://github.com/ballerina-platform/ballerinax-openapi-connectors"
+version = "1.1.0"
```
Comment (LakshanSS) [inline openapi/mft/Ballerina.toml:9]: Since this is a new connector, version can be `1.0.0`

Comment (SkNuwanTissera) [reply openapi/mft/Ballerina.toml]: Updated

**Review by aneeshafedo — APPROVED** [review]

**Pull request merged**